hoot-core/src/main/cpp/hoot/core/cmd/DiffConflateCmd.cpp

```diff
@@ -0,0 +1,258 @@
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2017 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+
+// Hoot
+#include <hoot/core/cmd/BaseCommand.h>
+#include <hoot/core/util/Factory.h>
+#include <hoot/core/util/MapProjector.h>
+#include <hoot/core/conflate/ConflateStatsHelper.h>
+#include <hoot/core/conflate/StatsComposer.h>
+#include <hoot/core/conflate/DiffConflator.h>
+#include <hoot/core/filters/StatusCriterion.h>
+#include <hoot/core/ops/BuildingOutlineUpdateOp.h>
+#include <hoot/core/ops/CalculateStatsOp.h>
+#include <hoot/core/ops/NamedOp.h>
+#include <hoot/core/ops/stats/IoSingleStat.h>
+#include <hoot/core/util/ConfigOptions.h>
+#include <hoot/core/io/OsmMapWriterFactory.h>
+#include <hoot/core/io/OsmXmlWriter.h>
+#include <hoot/core/io/MapStatsWriter.h>
+#include <hoot/core/io/ChangesetDeriver.h>
+#include <hoot/core/io/ElementSorter.h>
+#include <hoot/core/io/OsmXmlChangesetFileWriter.h>
+#include <hoot/core/util/Log.h>
+
+// Standard
+#include <fstream>
+
+// Tgs
+#include <tgs/System/Timer.h>
+
+#include <QFileInfo>
+
+using namespace std;
+using namespace Tgs;
+
+namespace hoot
+{
+
+class DiffConflateCmd : public BaseCommand
+{
+public:
+
+  static string className() { return "hoot::DiffConflateCmd"; }
+
+  DiffConflateCmd() { }
+
+  virtual QString getName() const { return "diff-conflate"; }
+
+  // Convenience function used when deriving a changeset
+  boost::shared_ptr<ChangesetDeriver> _sortInputs(QList<OsmMapPtr> inputMaps)
+  {
+    ElementSorterPtr sorted1(new ElementSorter(inputMaps[0]));
+    ElementSorterPtr sorted2(new ElementSorter(inputMaps[1]));
+    boost::shared_ptr<ChangesetDeriver> delta(new ChangesetDeriver(sorted1, sorted2));
+    return delta;
+  }
+
+  int runSimple(QStringList args)
+  {
+    Timer totalTime;
+    Timer t;
+    bool findChangeset = false;
+
+    QList<SingleStat> stats;
+    bool displayStats = false;
+    QString outputStatsFile;
+
+    // Handle Stats
+    if (args.endsWith("--stats"))
+    {
+      displayStats = true;
+      args.pop_back();
+    }
+
+    if (args.size() != 3)
+    {
+      cout << getHelp() << endl << endl;
+      throw HootException(QString("%1 takes three parameters.").arg(getName()));
+    }
+
+    // Decide if we need to create a changeset
+    if (args[2].endsWith("osc", Qt::CaseInsensitive))
+    {
+      findChangeset = true;
+    }
+
+    QString input1 = args[0];
+    QString input2 = args[1];
+    QString output = args[2];
+
+    LOG_INFO("Differentially conflating " << input1.right(50) << " with " << input2.right(50) <<
+             " and writing output to " << output.right(50));
+
+    double bytesRead = IoSingleStat(IoSingleStat::RChar).value;
+    LOG_VART(bytesRead);
+    QList< QList<SingleStat> > allStats;
+
+    // read input 1, save an original copy
+    OsmMapPtr originalMap(new OsmMap());
+    loadMap(originalMap, input1, ConfigOptions().getReaderConflateUseDataSourceIds1(), Status::Unknown1);
```

Comment:
It appears as though reading in the map (and the memory and time associated) is only needed when doing a changeset.  Therefore I think it should only be necessary within the following section at line 184:
```
    if (findChangeset)
    {
      // read input 1, save an original copy
      OsmMapPtr originalMap(new OsmMap());
      loadMap(originalMap, input1, ConfigOptions().getReaderConflateUseDataSourceIds1(), Status::Unknown1);
      // Calculate & write the changeset
      QList<OsmMapPtr> inputMaps;
      inputMaps.push_back(originalMap);
      inputMaps.push_back(result);
      OsmXmlChangesetFileWriter().write(output, _sortInputs(inputMaps));
    }
```
On larger datasets this can make a large difference if it is not needed and when it is needed, the scope of the OsmMap object is smaller.

Reply:
Good call. Thanks for the feedback!